From 94c707c156accc3c5e8d827b0c4f39e30954e45d Mon Sep 17 00:00:00 2001
From: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Date: Mon, 4 Oct 2021 17:11:59 +0300
Subject: [PATCH 255/322] [FIO toup] tools: image-host: print error messages to
 stderr

The make by default cuts off the stdout output from external tools,
so all error messages from the image-host are not shown in a make
output. Besides that, it is a common approach to use stderr stream
for error messages.
Use stderr for all error messages in image-host.

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 tools/image-host.c | 92 +++++++++++++++++++++++-----------------------
 1 file changed, 46 insertions(+), 46 deletions(-)

diff --git a/tools/image-host.c b/tools/image-host.c
index 33a224129a0..f3c36b6588a 100644
--- a/tools/image-host.c
+++ b/tools/image-host.c
@@ -35,7 +35,7 @@ static int fit_set_hash_value(void *fit, int noffset, uint8_t *value,
 
 	ret = fdt_setprop(fit, noffset, FIT_VALUE_PROP, value, value_len);
 	if (ret) {
-		printf("Can't set hash '%s' property for '%s' node(%s)\n",
+		fprintf(stderr, "Can't set hash '%s' property for '%s' node(%s)\n",
 		       FIT_VALUE_PROP, fit_get_name(fit, noffset, NULL),
 		       fdt_strerror(ret));
 		return ret == -FDT_ERR_NOSPACE ? -ENOSPC : -EIO;
@@ -69,20 +69,20 @@ static int fit_image_process_hash(void *fit, const char *image_name,
 	node_name = fit_get_name(fit, noffset, NULL);
 
 	if (fit_image_hash_get_algo(fit, noffset, &algo)) {
-		printf("Can't get hash algo property for '%s' hash node in '%s' image node\n",
+		fprintf(stderr, "Can't get hash algo property for '%s' hash node in '%s' image node\n",
 		       node_name, image_name);
 		return -ENOENT;
 	}
 
 	if (calculate_hash(data, size, algo, value, &value_len)) {
-		printf("Unsupported hash algorithm (%s) for '%s' hash node in '%s' image node\n",
+		fprintf(stderr, "Unsupported hash algorithm (%s) for '%s' hash node in '%s' image node\n",
 		       algo, node_name, image_name);
 		return -EPROTONOSUPPORT;
 	}
 
 	ret = fit_set_hash_value(fit, noffset, value, value_len);
 	if (ret) {
-		printf("Can't set hash value for '%s' hash node in '%s' image node\n",
+		fprintf(stderr, "Can't set hash value for '%s' hash node in '%s' image node\n",
 		       node_name, image_name);
 		return ret;
 	}
@@ -162,7 +162,7 @@ static int fit_image_setup_sig(struct image_sign_info *info,
 
 	node_name = fit_get_name(fit, noffset, NULL);
 	if (fit_image_hash_get_algo(fit, noffset, &algo_name)) {
-		printf("Can't get algo property for '%s' signature node in '%s' image node\n",
+		fprintf(stderr, "Can't get algo property for '%s' signature node in '%s' image node\n",
 		       node_name, image_name);
 		return -1;
 	}
@@ -181,7 +181,7 @@ static int fit_image_setup_sig(struct image_sign_info *info,
 	info->require_keys = require_keys;
 	info->engine_id = engine_id;
 	if (!info->checksum || !info->crypto) {
-		printf("Unsupported signature algorithm (%s) for '%s' signature node in '%s' image node\n",
+		fprintf(stderr, "Unsupported signature algorithm (%s) for '%s' signature node in '%s' image node\n",
 		       algo_name, node_name, image_name);
 		return -1;
 	}
@@ -229,7 +229,7 @@ static int fit_image_process_sig(const char *keydir, void *keydest,
 	region.size = size;
 	ret = info.crypto->sign(&info, &region, 1, &value, &value_len);
 	if (ret) {
-		printf("Failed to sign '%s' signature node in '%s' image node: %d\n",
+		fprintf(stderr, "Failed to sign '%s' signature node in '%s' image node: %d\n",
 		       node_name, image_name, ret);
 
 		/* We allow keys to be missing */
@@ -243,7 +243,7 @@ static int fit_image_process_sig(const char *keydir, void *keydest,
 	if (ret) {
 		if (ret == -FDT_ERR_NOSPACE)
 			return -ENOSPC;
-		printf("Can't write signature for '%s' signature node in '%s' conf node: %s\n",
+		fprintf(stderr, "Can't write signature for '%s' signature node in '%s' conf node: %s\n",
 		       node_name, image_name, fdt_strerror(ret));
 		return -1;
 	}
@@ -260,7 +260,7 @@ static int fit_image_process_sig(const char *keydir, void *keydest,
 	if (keydest) {
 		ret = info.crypto->add_verify_data(&info, keydest);
 		if (ret) {
-			printf("Failed to add verification data for '%s' signature node in '%s' image node\n",
+			fprintf(stderr, "Failed to add verification data for '%s' signature node in '%s' image node\n",
 			       node_name, image_name);
 			return ret;
 		}
@@ -279,21 +279,21 @@ static int fit_image_read_data(char *filename, unsigned char *data,
 	/* Open file */
 	fd = open(filename, O_RDONLY | O_BINARY);
 	if (fd < 0) {
-		printf("Can't open file %s (err=%d => %s)\n",
+		fprintf(stderr, "Can't open file %s (err=%d => %s)\n",
 		       filename, errno, strerror(errno));
 		return -1;
 	}
 
 	/* Compute file size */
 	if (fstat(fd, &sbuf) < 0) {
-		printf("Can't fstat file %s (err=%d => %s)\n",
+		fprintf(stderr, "Can't fstat file %s (err=%d => %s)\n",
 		       filename, errno, strerror(errno));
 		goto err;
 	}
 
 	/* Check file size */
 	if (sbuf.st_size != expected_size) {
-		printf("File %s don't have the expected size (size=%lld, expected=%d)\n",
+		fprintf(stderr, "File %s don't have the expected size (size=%lld, expected=%d)\n",
 		       filename, (long long)sbuf.st_size, expected_size);
 		goto err;
 	}
@@ -301,14 +301,14 @@ static int fit_image_read_data(char *filename, unsigned char *data,
 	/* Read data */
 	n = read(fd, data, sbuf.st_size);
 	if (n < 0) {
-		printf("Can't read file %s (err=%d => %s)\n",
+		fprintf(stderr, "Can't read file %s (err=%d => %s)\n",
 		       filename, errno, strerror(errno));
 		goto err;
 	}
 
 	/* Check that we have read all the file */
 	if (n != sbuf.st_size) {
-		printf("Can't read all file %s (read %zd bytes, expexted %lld)\n",
+		fprintf(stderr, "Can't read all file %s (read %zd bytes, expexted %lld)\n",
 		       filename, n, (long long)sbuf.st_size);
 		goto err;
 	}
@@ -327,14 +327,14 @@ static int get_random_data(void *data, int size)
 	int i, ret = 0;
 
 	if (!tmp) {
-		printf("%s: pointer data is NULL\n", __func__);
+		fprintf(stderr, "%s: pointer data is NULL\n", __func__);
 		ret = -1;
 		goto out;
 	}
 
 	ret = clock_gettime(CLOCK_MONOTONIC, &date);
 	if (ret < 0) {
-		printf("%s: clock_gettime has failed (err=%d, str=%s)\n",
+		fprintf(stderr, "%s: clock_gettime has failed (err=%d, str=%s)\n",
 		       __func__, ret, strerror(errno));
 		goto out;
 	}
@@ -360,7 +360,7 @@ static int fit_image_setup_cipher(struct image_cipher_info *info,
 	int ret = -1;
 
 	if (fit_image_cipher_get_algo(fit, noffset, &algo_name)) {
-		printf("Can't get algo name for cipher in image '%s'\n",
+		fprintf(stderr, "Can't get algo name for cipher in image '%s'\n",
 		       image_name);
 		goto out;
 	}
@@ -370,7 +370,7 @@ static int fit_image_setup_cipher(struct image_cipher_info *info,
 	/* Read the key name */
 	info->keyname = fdt_getprop(fit, noffset, FIT_KEY_HINT, NULL);
 	if (!info->keyname) {
-		printf("Can't get key name for cipher in image '%s'\n",
+		fprintf(stderr, "Can't get key name for cipher in image '%s'\n",
 		       image_name);
 		goto out;
 	}
@@ -389,7 +389,7 @@ static int fit_image_setup_cipher(struct image_cipher_info *info,
 
 	info->cipher = image_get_cipher_algo(algo_name);
 	if (!info->cipher) {
-		printf("Can't get algo for cipher '%s'\n", image_name);
+		fprintf(stderr, "Can't get algo for cipher '%s'\n", image_name);
 		goto out;
 	}
 
@@ -398,7 +398,7 @@ static int fit_image_setup_cipher(struct image_cipher_info *info,
 		 info->keydir, info->keyname, ".bin");
 	info->key = malloc(info->cipher->key_len);
 	if (!info->key) {
-		printf("Can't allocate memory for key\n");
+		fprintf(stderr, "Can't allocate memory for key\n");
 		ret = -1;
 		goto out;
 	}
@@ -409,7 +409,7 @@ static int fit_image_setup_cipher(struct image_cipher_info *info,
 
 	info->iv = malloc(info->cipher->iv_len);
 	if (!info->iv) {
-		printf("Can't allocate memory for iv\n");
+		fprintf(stderr, "Can't allocate memory for iv\n");
 		ret = -1;
 		goto out;
 	}
@@ -443,7 +443,7 @@ int fit_image_write_cipher(void *fit, int image_noffset, int noffset,
 		goto out;
 	}
 	if (ret) {
-		printf("Can't replace data with ciphered data (err = %d)\n", ret);
+		fprintf(stderr, "Can't replace data with ciphered data (err = %d)\n", ret);
 		goto out;
 	}
 
@@ -454,7 +454,7 @@ int fit_image_write_cipher(void *fit, int image_noffset, int noffset,
 		goto out;
 	}
 	if (ret) {
-		printf("Can't add unciphered data size (err = %d)\n", ret);
+		fprintf(stderr, "Can't add unciphered data size (err = %d)\n", ret);
 		goto out;
 	}
 
@@ -494,7 +494,7 @@ fit_image_process_cipher(const char *keydir, void *keydest, void *fit,
 	if (keydest) {
 		ret = info.cipher->add_cipher_data(&info, keydest, fit, node_noffset);
 		if (ret) {
-			printf("Failed to add verification data for cipher '%s' in image '%s'\n",
+			fprintf(stderr, "Failed to add verification data for cipher '%s' in image '%s'\n",
 			       info.keyname, image_name);
 			goto out;
 		}
@@ -524,13 +524,13 @@ int fit_image_cipher_data(const char *keydir, void *keydest,
 	/* Get image name */
 	image_name = fit_get_name(fit, image_noffset, NULL);
 	if (!image_name) {
-		printf("Can't get image name\n");
+		fprintf(stderr, "Can't get image name\n");
 		return -1;
 	}
 
 	/* Get image data and data length */
 	if (fit_image_get_data(fit, image_noffset, &data, &size)) {
-		printf("Can't get image data/size\n");
+		fprintf(stderr, "Can't get image data/size\n");
 		return -1;
 	}
 
@@ -544,7 +544,7 @@ int fit_image_cipher_data(const char *keydir, void *keydest,
 	if (fdt_getprop(fit, image_noffset, "data-size-unciphered", &len))
 		return 0;
 	if (len != -FDT_ERR_NOTFOUND) {
-		printf("Failure testing for data-size-unciphered\n");
+		fprintf(stderr, "Failure testing for data-size-unciphered\n");
 		return -1;
 	}
 
@@ -554,7 +554,7 @@ int fit_image_cipher_data(const char *keydir, void *keydest,
 	if (cipher_node_offset == -FDT_ERR_NOTFOUND)
 		return 0;
 	if (cipher_node_offset < 0) {
-		printf("Failure getting cipher node\n");
+		fprintf(stderr, "Failure getting cipher node\n");
 		return -1;
 	}
 	if (!IMAGE_ENABLE_ENCRYPT || !keydir)
@@ -609,7 +609,7 @@ int fit_image_add_verification_data(const char *keydir, void *keydest,
 
 	/* Get image data and data length */
 	if (fit_image_get_data(fit, image_noffset, &data, &size)) {
-		printf("Can't get image data/size\n");
+		fprintf(stderr, "Can't get image data/size\n");
 		return -1;
 	}
 
@@ -736,7 +736,7 @@ static int fit_config_add_hash(void *fit, const char *conf_name, const char *sig
 	}
 
 	if (!hash_count) {
-		printf("Failed to find any hash nodes in configuration '%s/%s' image '%s' - without these it is not possible to verify this image\n",
+		fprintf(stderr, "Failed to find any hash nodes in configuration '%s/%s' image '%s' - without these it is not possible to verify this image\n",
 		       conf_name, sig_name, iname);
 		return -ENOMSG;
 	}
@@ -746,7 +746,7 @@ static int fit_config_add_hash(void *fit, const char *conf_name, const char *sig
 				     FIT_CIPHER_NODENAME);
 	if (noffset != -FDT_ERR_NOTFOUND) {
 		if (noffset < 0) {
-			printf("Failed to get cipher node in configuration '%s/%s' image '%s': %s\n",
+			fprintf(stderr, "Failed to get cipher node in configuration '%s/%s' image '%s': %s\n",
 			       conf_name, sig_name, iname,
 			       fdt_strerror(noffset));
 			return -EIO;
@@ -761,12 +761,12 @@ static int fit_config_add_hash(void *fit, const char *conf_name, const char *sig
 	return 0;
 
 err_mem:
-	printf("Out of memory processing configuration '%s/%s'\n", conf_name,
+	fprintf(stderr, "Out of memory processing configuration '%s/%s'\n", conf_name,
 	       sig_name);
 	return -ENOMEM;
 
 err_path:
-	printf("Failed to get path for image '%s' in configuration '%s/%s': %s\n",
+	fprintf(stderr, "Failed to get path for image '%s' in configuration '%s/%s': %s\n",
 	       iname, conf_name, sig_name, fdt_strerror(ret));
 	return -ENOENT;
 }
@@ -814,7 +814,7 @@ static int fit_config_get_hash_list(void *fit, int conf_noffset,
 								     iname, index);
 
 			if (image_noffset < 0) {
-				printf("Failed to find image '%s' in  configuration '%s/%s'\n",
+				fprintf(stderr, "Failed to find image '%s' in  configuration '%s/%s'\n",
 				       iname, conf_name, sig_name);
 				if (allow_missing)
 					continue;
@@ -833,7 +833,7 @@ static int fit_config_get_hash_list(void *fit, int conf_noffset,
 	}
 
 	if (!image_count) {
-		printf("Failed to find any images for configuration '%s/%s'\n",
+		fprintf(stderr, "Failed to find any images for configuration '%s/%s'\n",
 		       conf_name, sig_name);
 		return -ENOMSG;
 	}
@@ -841,7 +841,7 @@ static int fit_config_get_hash_list(void *fit, int conf_noffset,
 	return 0;
 
 err_mem:
-	printf("Out of memory processing configuration '%s/%s'\n", conf_name,
+	fprintf(stderr, "Out of memory processing configuration '%s/%s'\n", conf_name,
 	       sig_name);
 	return -ENOMEM;
 }
@@ -875,12 +875,12 @@ static int fit_config_get_data(void *fit, int conf_noffset, int noffset,
 			fdt_regions, ARRAY_SIZE(fdt_regions),
 			path, sizeof(path), 1);
 	if (count < 0) {
-		printf("Failed to hash configuration '%s/%s': %s\n", conf_name,
+		fprintf(stderr, "Failed to hash configuration '%s/%s': %s\n", conf_name,
 		       sig_name, fdt_strerror(ret));
 		return -EIO;
 	}
 	if (count == 0) {
-		printf("No data to hash for configuration '%s/%s': %s\n",
+		fprintf(stderr, "No data to hash for configuration '%s/%s': %s\n",
 		       conf_name, sig_name, fdt_strerror(ret));
 		return -EINVAL;
 	}
@@ -888,7 +888,7 @@ static int fit_config_get_data(void *fit, int conf_noffset, int noffset,
 	/* Build our list of data blocks */
 	region = fit_region_make_list(fit, fdt_regions, count, NULL);
 	if (!region) {
-		printf("Out of memory hashing configuration '%s/%s'\n",
+		fprintf(stderr, "Out of memory hashing configuration '%s/%s'\n",
 		       conf_name, sig_name);
 		return -ENOMEM;
 	}
@@ -901,7 +901,7 @@ static int fit_config_get_data(void *fit, int conf_noffset, int noffset,
 	}
 	region_prop = malloc(len);
 	if (!region_prop) {
-		printf("Out of memory setting up regions for configuration '%s/%s'\n",
+		fprintf(stderr, "Out of memory setting up regions for configuration '%s/%s'\n",
 		       conf_name, sig_name);
 		return -ENOMEM;
 	}
@@ -946,7 +946,7 @@ static int fit_config_process_sig(const char *keydir, void *keydest,
 				&value_len);
 	free(region);
 	if (ret) {
-		printf("Failed to sign '%s' signature node in '%s' conf node\n",
+		fprintf(stderr, "Failed to sign '%s' signature node in '%s' conf node\n",
 		       node_name, conf_name);
 
 		/* We allow keys to be missing */
@@ -960,7 +960,7 @@ static int fit_config_process_sig(const char *keydir, void *keydest,
 	if (ret) {
 		if (ret == -FDT_ERR_NOSPACE)
 			return -ENOSPC;
-		printf("Can't write signature for '%s' signature node in '%s' conf node: %s\n",
+		fprintf(stderr, "Can't write signature for '%s' signature node in '%s' conf node: %s\n",
 		       node_name, conf_name, fdt_strerror(ret));
 		return -1;
 	}
@@ -974,7 +974,7 @@ static int fit_config_process_sig(const char *keydir, void *keydest,
 	if (keydest) {
 		ret = info.crypto->add_verify_data(&info, keydest);
 		if (ret) {
-			printf("Failed to add verification data for '%s' signature node in '%s' configuration node\n",
+			fprintf(stderr, "Failed to add verification data for '%s' signature node in '%s' configuration node\n",
 			       node_name, conf_name);
 		}
 		return ret;
@@ -1024,7 +1024,7 @@ int fit_cipher_data(const char *keydir, void *keydest, void *fit,
 	/* Find images parent node offset */
 	images_noffset = fdt_path_offset(fit, FIT_IMAGES_PATH);
 	if (images_noffset < 0) {
-		printf("Can't find images parent node '%s' (%s)\n",
+		fprintf(stderr, "Can't find images parent node '%s' (%s)\n",
 		       FIT_IMAGES_PATH, fdt_strerror(images_noffset));
 		return images_noffset;
 	}
@@ -1059,7 +1059,7 @@ int fit_add_verification_data(const char *keydir, void *keydest, void *fit,
 	/* Find images parent node offset */
 	images_noffset = fdt_path_offset(fit, FIT_IMAGES_PATH);
 	if (images_noffset < 0) {
-		printf("Can't find images parent node '%s' (%s)\n",
+		fprintf(stderr, "Can't find images parent node '%s' (%s)\n",
 		       FIT_IMAGES_PATH, fdt_strerror(images_noffset));
 		return images_noffset;
 	}
@@ -1086,7 +1086,7 @@ int fit_add_verification_data(const char *keydir, void *keydest, void *fit,
 	/* Find configurations parent node offset */
 	confs_noffset = fdt_path_offset(fit, FIT_CONFS_PATH);
 	if (confs_noffset < 0) {
-		printf("Can't find images parent node '%s' (%s)\n",
+		fprintf(stderr, "Can't find images parent node '%s' (%s)\n",
 		       FIT_CONFS_PATH, fdt_strerror(confs_noffset));
 		return -ENOENT;
 	}
-- 
2.34.1

